invenio_rdm_records/services/pids/providers/datacite.py

```diff
@@ -111,9 +111,12 @@ def _log_errors(errors):
         # DataCiteError is a tuple with the errors on the first
         errors = json.loads(errors.args[0])["errors"]
         for error in errors:
-            field = error["source"]
+            field = error.get("source", None)
             reason = error["title"]
-            current_app.logger.warning(f"Error in {field}: {reason}")
+            if field:
+                current_app.logger.warning(f"Error in {field}: {reason}")
+            else:
+                current_app.logger.warning(f"Error: {reason}")
```

Comment:
how about this:
```suggestion
            if field:
                current_app.logger.warning(f"Error in {field if field else 'Missing field'}: {reason}")
            else:
                current_app.logger.warning(f"Error: {reason}")
```
to avoid adding conditions only to log warnings ?